### PR TITLE
Addition of event dux module

### PR DESCRIPTION
**Pull request by prodhan29**
- add dux file of events.
- add test code for events.

### CODE REVIEW

File: src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -147,4 +147,4 @@ exports[`components/page/SideMenu sets the collapsedModuleLabel correctly from c
                               key=\\"child-nav-0\\"
```
Comment (p-netm) [inline src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap]: I thin k that this file does not seem to belong in this pr.

File: src/store/ducks/tests/events.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  getEventsArray,
+  reducerName as eventReducerName,
+} from '../events';
+
+import * as fixtures from '../tests/fixtures';
+
+reducerRegistry.register(eventReducerName, eventReducer);
+
+describe('reducers/clients', () => {
+  beforeEach(() => {
```
Comment (p-netm) [inline src/store/ducks/tests/events.test.ts]: maybe for the dux's modules tests; you can follow this checklist, its not guaranteed to fully exercise the code but i think it would improve the dux's testing especially for this test file.

```typescript
/**
 *  selectors:
 *       - on initial state,
 *       - on a non-empty state,
 *  reducers:
 *          - response towards invalid/unknown type
 *      - for each action:
 *          - use selectors to check state after add action
 *              - when the starting-state is empty
 *              - when the starting-state has stuff.
 *          - response towards remove action.
 *
 */
```

File: src/store/ducks/tests/fixtures.ts [hunk 3]
```diff
@@ -345,5 +346,40 @@ export const household3: Household = {
...
+  serverVersion: 1552287303783,
+  team: '',
+  teamId: '',
+  type: 'Event',
+  version: 1552287303606,
+};
+
 export const clients: Client[] = [client1, client2, client3, client4, client5, client6, client7];
 export const households: Household[] = [household1, household2, household3];
+export const events: Event[] = [event1];
```
Comment (p-netm) [inline src/store/ducks/tests/fixtures.ts]: If you are to follow the above checklist, you will most definitely find that you need more than one event for your fixtures.

File: src/store/ducks/events.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      });
+
+    default:
+      return state;
+  }
+}
+
+// Selectors
+
+/** returns the events of the event in the store as array
```
Comment (p-netm) [inline src/store/ducks/events.ts]: nitpick: how do you feel about just `returns an array of events`

File: src/store/ducks/events.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ * @param {Partial<Store>} state - the redux store
+ * @return { Event } - events array
+ */
+export function getEventsArray(state: Partial<Store>): Event[] {
+  return values((state as any)[reducerName].eventsById);
+}
+
+/** returns the events of the event in the store by their ids
```
Comment (p-netm) [inline src/store/ducks/events.ts]: also for this, the `events of the event` is a tad confusing.

File: src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap (hunk 1 above)
Comment (prodhan29) [reply src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap]: i accidentally updated the snapshot. but it is as same as in the master branch.

File: src/store/ducks/events.ts (hunk 4 above)
Comment (prodhan29) [reply src/store/ducks/events.ts]: Fixed it.

File: src/store/ducks/events.ts (hunk 5 above)
Comment (prodhan29) [reply src/store/ducks/events.ts]: Fixed It

File: src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap (hunk 1 above)
Comment (p-netm) [reply src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap]: @prodhan29 if its not to much trouble please remove it.

File: src/store/ducks/events.ts (hunk 4 above)
Comment (prodhan29) [reply src/store/ducks/events.ts]: renamed it.

File: src/store/ducks/tests/fixtures.ts (hunk 3 above)
Comment (prodhan29) [reply src/store/ducks/tests/fixtures.ts]: Added more test code as you suggested

File: src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap (hunk 1 above)
Comment (prodhan29) [reply src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap]: this branch has already 5 commits. and this file was commited in middle of it. can you please suggest how can i remove it safely.

**Comment by p-netm** [comment]
@prodhan29 Add a removeEventsAction to make the tests pass, its usually very helpful when testing not only in the dux but also components that use this dux module.

File: src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap (hunk 1 above)
Comment (p-netm) [reply src/components/page/SideMenu/tests/__snapshots__/index.test.tsx.snap]: ill handle it,

File: src/store/ducks/tests/events.test.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { values } from 'lodash';
```
Comment (p-netm) [inline src/store/ducks/tests/events.test.ts]: unused import

File: src/store/ducks/tests/events.test.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
+import reducerRegistry from '@onaio/redux-reducer-registry';
```
Comment (moshthepitt) [inline src/store/ducks/tests/events.test.ts]: @p-netm do we have a test in which we verify that adding new events does not replace existing ones?

Comment (p-netm) [reply src/store/ducks/tests/events.test.ts]: @moshthepitt yes, [here](https://github.com/OpenSRP/opensrp-web/blob/7aa2f4ed487bd232f85f5659a99e891c09342fee/src/store/ducks/tests/events.test.ts#L46)

File: clients/core/src/store/ducks/baseDux/README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+/** reducer name for the ANC module */
+export const reducerName = 'opensrp-web/client-type/ANC';
+
+/** ANC Reducer */
+const reducer = reducerFactory<ANCClientType>(reducerName);
+
+// action
+/** actionCreator returns action to to add anc records to store */
+export const fetchANC = fetchActionCreatorFactory<ANCClientType, 'baseEntityId'>(reducerName);
```
Comment (p-netm) [inline clients/core/src/store/ducks/baseDux/README.md]: @prodhan29 please help me fix this, `baseEntityid` should have been passed inside the function and not as a generic type

Comment (prodhan29) [reply clients/core/src/store/ducks/baseDux/README.md]: Fixed

**Comment by prodhan29** [comment]
> `fetchFactory` => `fetchItemsCreatorFactory`
> `removeFactory` => `removeItemsCreatorFactory`
> 
> And remember to update Readme

As you have renamed the `fetchFactory` to `fetchCreatorFactory` and similarly I changed the `removeFactory` to `removeActionCreatorFactory`.

File: clients/core/src/store/ducks/events.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import {
+    fetchActionCreatorFactory,
+    removeActionCreatorFactory,
+    setTotalRecordsFactory,
+    reducerFactory,
+    getItemsByIdFactory,
+    getItemsArrayFactory,
+    getItemByIdFactory,
+    getTotalRecordsFactory,
+} from './baseDux';
```
Comment (p-netm) [inline clients/core/src/store/ducks/events.ts]: setTotalRecords and getTotalRecords are unused, please remove them if they do not apply to this module.

**Comment by p-netm** [comment]
superseded by #195
